### PR TITLE
[Feature-14839][Alert]Add Webhook Alert Plugin to support Enterprise WeChat notifications by robot

### LINKED ISSUE #14839: [Feature][Alert]Enterprise WeChat support send notifications by robot
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Chinese：经与企业微信论坛和官方客服确认，企业微信无法发送告警信息到已有的内部群，原因是无法获取到已经创建内部群的chatid，只能通过接口创建，必须要单独创建群聊，不满足实际需求。论坛上有许多人反映此问题，且经受困扰。同时，DolphinScheduler的贡献者群内也有相同的诉求。该问题亟待解决。
English: After confirmation with the enterprise WeChat forum and official customer service, it was found that enterprise WeChat cannot send alarm information to existing internal groups because it cannot obtain the chatid of the already created internal group. It can only be created through the interface and must be created separately, which does not meet actual needs. Many people on the forum have reported this issue and are experiencing difficulties. At the same time, there are similar demands within the contributor group of DolphinScheduler. This problem urgently needs to be solved.

### Use case

经调研，可以通过在群内增加机器人的方式解决。我会新增加一种告警插件，以满足这个需求。
After research, it can be solved by adding robots to the group. I will add a new alarm plugin to meet this requirement.

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ahuljh**
Add Webhook Alert Plugin to support Enterprise WeChat notifications by robot
close #14839 

## Purpose of the pull request
The purpose of this PR is to support Enterprise WeChat notifications by robot. This pull request adds webhook alert plugin.

## Brief change log

- Add dolphinscheduler-alert-webhook to the module dolphinscheduler-alert-plugins.
- Add classes on which alarms rely, including WebHookSender, WebHookAlertChannel, WebHookAlertChannelFactory, WebHookAlertConstants, and WebHookAlertMessage.
- Modify the pom.xml file of the dolphinscheduler alert plugins module.
- Add pom.xml file for dolphinscheduler alert webhook module and introduce corresponding dependencies
- Write test classes and methods required for message sending.

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Comment by caishunfeng** [comment]
Hi @ahuljh Please add the documents of this new feature.

**Comment by ahuljh** [comment]
> Hi @ahuljh Please add the documents of this new feature.

Thank you for your reminder. I will supplement the guidance document, verify the code is correct, and then upload it again. After all have been uploaded, I will remind you to check again. Thanks.

**Comment by ahuljh** [comment]
> Hi @ahuljh Please add the documents of this new feature.

Hi  @caishunfeng The documents and code have been supplemented and verified. Could you please continue with the merge process? Thanks.

**Comment by qingwli** [comment]
About this feature, we already have `http` and `wechat` plugin, this webhook looks like another way to use wechat by HTTP. How about we modify HTTP plugin and add a simple way to use HTTP, the same as just a HTTP URL or we modify wechat plugin to support this way?

**Comment by qingwli** [comment]
image
And wechat webhook can @ someone, maybe we modify wechat plugin is better

**Comment by ahuljh** [comment]
> 



> image And wechat webhook can @ someone, maybe we modify wechat plugin is better

In fact, we conducted an evaluation before adding the webhook plugin, and if we add the webhook method to the wechat plugin, the Secret and CropId options will be redundant and cannot be adapted based on this. If using the Http plugin, it is necessary to modify the source code of the Http plugin, which will affect the use of other alerts through HTTP. After a comprehensive decision, we have decided to add the Webhook method. This is also the cheapest way.

**Comment by ahuljh** [comment]
Hi, @qingwli I will consider adding options for @ everyone and @ designated person. If additional options are needed, I will resubmit the Pull Request.
In addition, the goal of this webhook plugin is to adapt to various platforms, not just enterprise WeChat. This plugin will adapt to various other places that require alerting through webhook.
Thank you for your suggestion.

**Comment by qingwli** [comment]
About Wechat plugin, we add a send type webhook and add an input field, let the Secret and CropId options not required is a good way? We can support two WeChat way and adapt pervious way

**Comment by ahuljh** [comment]
> About Wechat plugin, we add a send type webhook and add an input field, let the Secret and CropId options not required is a good way? We can support two WeChat way and adapt pervious way

By modifying WeChat plugins, you can also meet your desired needs for WeChat. But is it no longer supported for other places that require inputting webhooks. Through webhook plugin, it can be flexibly applied to other places, rather than just WeChat.

**Comment by qingwli** [comment]
About webhook plugin can support different webhook request, but like wechat have @ function, the field is not common in different webhook request, so it's different to be a common plugin if you add some config. If do that it's similar to HTTP plugin.


In my opinion, for ds we support so many plugins, and for any alert ways always have different request combinations, The root way is HTTP too, we just support usage way for easy use. And for user side, If I want to use wechat , I always click Wechat button not webhook. So I think we should enhance wechat plugin.

**Comment by ahuljh** [comment]
Hi, @caishunfeng @zhongjiajie @EricGao888 My company has already used this method, and many people have this need. Do you think it is necessary to add webhook plugins to adapt to different platforms? Or is it to modify the WeChat plugin to only meet the needs of WeChat?

**Comment by qingwli** [comment]
cc @SbloodyS

**Comment by ruanwenjun** [comment]
It's better to rename it to WeChat-webhook, rather than webhook. Not all webhooks can support markdown.

**Comment by ahuljh** [comment]
> It's better to rename it to WeChat-webhook, rather than webhook. Not all webhooks can support markdown.

Yes, your opinion is very good. I will rename the plugin this week.

**Comment by ahuljh** [comment]
Expect my code to be quickly merged

**Comment by fuchanghai** [comment]
@caishunfeng @ruanwenjun plz run CI

**Comment by ocean0551** [comment]
还没有合并吗？

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-wechat-webhook/src/main/java/org/apache/dolphinscheduler/plugin/alert/webhook/WeChatWebHookMessage.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.plugin.alert.webhook;
+
+import org.apache.dolphinscheduler.alert.api.ShowType;
+
+import java.util.Map;
+
+public class WeChatWebHookMessage {
```
Comment (SbloodyS) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-wechat-webhook/src/main/java/org/apache/dolphinscheduler/plugin/alert/webhook/WeChatWebHookMessage.java:24]: Please use `@Data` to generate getter and setter.

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-wechat-webhook/src/main/java/org/apache/dolphinscheduler/plugin/alert/webhook/WeChatWebHookSender.java [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        }
+
+        Map<String, String> contentMap = new HashMap<>();
+        contentMap.put(WeChatWebHookAlertConstants.CONTENT_KEY, data);
+        WeChatWebHookMessage weChatWebHookMessage = new WeChatWebHookMessage(showType, contentMap);
+        String msgJson = JSONUtils.toJsonString(weChatWebHookMessage);
+        try {
+            return checkSendResult(post(webHookURL, msgJson));
+        } catch (Exception e) {
+            logger.info("send webhook alert msg exception : {}", e.getMessage());
```
Comment (SbloodyS) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-wechat-webhook/src/main/java/org/apache/dolphinscheduler/plugin/alert/webhook/WeChatWebHookSender.java:78]:
```suggestion
            logger.warn("send webhook alert msg exception : {}", e.getMessage());
```

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-wechat-webhook/src/main/java/org/apache/dolphinscheduler/plugin/alert/webhook/WeChatWebHookSender.java [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+            CloseableHttpResponse response = httpClient.execute(httpPost);
+            String resp;
+            try {
+                HttpEntity entity = response.getEntity();
+                resp = EntityUtils.toString(entity, WeChatWebHookAlertConstants.CHARSET);
+                EntityUtils.consume(entity);
+            } finally {
+                response.close();
+            }
+            logger.info("WebHook send [{}], param:{}, resp:{}",
```
Comment (SbloodyS) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-wechat-webhook/src/main/java/org/apache/dolphinscheduler/plugin/alert/webhook/WeChatWebHookSender.java:99]:
```suggestion
            logger.info("webhook post url: {}, param:{}, resp:{}",
```

**Review by SbloodyS — COMMENTED** [review]
Please fix CI.

**Comment by SbloodyS** [comment]
Are you still working on this? @ahuljh

**Pull request closed without merging**